src/main/java/com/google/devtools/build/lib/rules/cpp/CcLibrary.java

```diff
@@ -327,7 +327,11 @@ public static void init(
     if (ruleContext.getRule().getImplicitOutputsFunction() != ImplicitOutputsFunction.NONE
         || !ccCompilationOutputs.isEmpty()) {
       if (featureConfiguration.isEnabled(CppRuleClasses.TARGETS_WINDOWS)) {
-        String dllNameSuffix = CppHelper.getDLLHashSuffix(ruleContext, featureConfiguration);
+        Artifact customDefFile = common.getWinDefFile();
+        String dllNameSuffix =
+            customDefFile != null
+                ? CppHelper.getDLLHashSuffix(ruleContext, featureConfiguration)
```

Comment:
Reversed logic? And also, can you put this logic in `getDLLHashSuffix` function? It's used at other places as well.

Reply:
For some reasons, I did not realize the reversed logic. Thank you for pointing it out.